### PR TITLE
Renamed AsyncRequestProvider to AsyncRequestBody to better match sync's RequestBody.

### PR DESCRIPTION
**Pull request by millems**

### CODE REVIEW

File: codegen/src/test/resources/software/amazon/awssdk/codegen/poet/client/test-async-client-class.java [hunk 1]
```diff
@@ -434,7 +434,7 @@ public PaginatedOperationWithoutResultKeyPublisher paginatedOperationWithoutResu
      */
     @Override
     public CompletableFuture<StreamingInputOperationResponse> streamingInputOperation(
-            StreamingInputOperationRequest streamingInputOperationRequest, AsyncRequestProvider requestProvider) {
+            StreamingInputOperationRequest streamingInputOperationRequest, AsyncRequestBody requestProvider) {
```
Comment (shorea) [inline codegen/src/test/resources/software/amazon/awssdk/codegen/poet/client/test-async-client-class.java]: param name

File: core/src/main/java/software/amazon/awssdk/core/async/AsyncRequestBody.java [hunk 2]
```diff
@@ -41,49 +41,48 @@
...
     /**
      * @return The content length of the data being produced.
      */
     long contentLength();
 
     /**
-     * Creates an {@link AsyncRequestProvider} that produces data from the contents of a file. See
-     * {@link FileAsyncRequestProvider#builder} to create a customized provider implementation.
+     * Creates an {@link AsyncRequestBody} that produces data from the contents of a file. See
+     * {@link FileAsyncRequestBody#builder} to create a customized provider implementation.
```
Comment (shorea) [inline core/src/main/java/software/amazon/awssdk/core/async/AsyncRequestBody.java]: provider

File: codegen/src/main/java/software/amazon/awssdk/codegen/docs/AsyncOperationDocProvider.java [hunk 3]
```diff
@@ -28,7 +28,7 @@
     private static final String REQUEST_PROVIDER_DOCS =
```
Comment (shorea) [inline codegen/src/main/java/software/amazon/awssdk/codegen/docs/AsyncOperationDocProvider.java]: provider

File: codegen/src/main/java/software/amazon/awssdk/codegen/poet/client/AsyncClientInterface.java [hunk 4]
```diff
@@ -269,7 +269,7 @@ private MethodSpec traditionalMethod(OperationModel opModel) {
                 .addJavadoc(opModel.getDocs(model, ClientType.ASYNC));
 
         if (opModel.hasStreamingInput()) {
-            builder.addParameter(ClassName.get(AsyncRequestProvider.class), "requestProvider");
+            builder.addParameter(ClassName.get(AsyncRequestBody.class), "requestProvider");
```
Comment (shorea) [inline codegen/src/main/java/software/amazon/awssdk/codegen/poet/client/AsyncClientInterface.java]: provider

File: codegen/src/main/java/software/amazon/awssdk/codegen/poet/client/specs/JsonProtocolSpec.java [hunk 5]
```diff
@@ -155,13 +155,13 @@ public CodeBlock asyncExecutionHandler(OperationModel opModel) {
         ClassName requestType = poetExtensions.getModelClass(opModel.getInput().getVariableType());
         ClassName marshaller = poetExtensions.getRequestTransformClass(opModel.getInputShape().getShapeName() + "Marshaller");
 
-        String asyncRequestProvider = opModel.hasStreamingInput() ? ".withAsyncRequestProvider(requestProvider)"
+        String asyncRequestBody = opModel.hasStreamingInput() ? ".withAsyncRequestBody(requestProvider)"
```
Comment (shorea) [inline codegen/src/main/java/software/amazon/awssdk/codegen/poet/client/specs/JsonProtocolSpec.java]: provider

File: codegen/src/main/java/software/amazon/awssdk/codegen/poet/client/specs/QueryXmlProtocolSpec.java [hunk 6]
```diff
@@ -152,13 +152,13 @@ public CodeBlock asyncExecutionHandler(OperationModel opModel) {
         ClassName requestType = poetExtensions.getModelClass(opModel.getInput().getVariableType());
         ClassName marshaller = poetExtensions.getRequestTransformClass(opModel.getInputShape().getShapeName() + "Marshaller");
 
-        String asyncRequestProvider = opModel.hasStreamingInput() ? ".withAsyncRequestProvider(requestProvider)"
+        String asyncRequestBody = opModel.hasStreamingInput() ? ".withAsyncRequestBody(requestProvider)"
```
Comment (shorea) [inline codegen/src/main/java/software/amazon/awssdk/codegen/poet/client/specs/QueryXmlProtocolSpec.java]: provider

File: codegen/src/test/resources/software/amazon/awssdk/codegen/poet/client/test-async-client-class.java [hunk 7]
```diff
@@ -413,7 +413,7 @@ public PaginatedOperationWithoutResultKeyPublisher paginatedOperationWithoutResu
      * @param streamingInputOperationRequest
      * @param requestProvider
```
Comment (shorea) [inline codegen/src/test/resources/software/amazon/awssdk/codegen/poet/client/test-async-client-class.java]: param name

File: codegen/src/test/resources/software/amazon/awssdk/codegen/poet/client/test-json-async-client-interface.java [hunk 8]
```diff
@@ -724,7 +724,7 @@ default PaginatedOperationWithoutResultKeyPublisher paginatedOperationWithoutRes
      *      target="_top">AWS API Documentation</a>
      */
     default CompletableFuture<StreamingInputOperationResponse> streamingInputOperation(
-            StreamingInputOperationRequest streamingInputOperationRequest, AsyncRequestProvider requestProvider) {
+            StreamingInputOperationRequest streamingInputOperationRequest, AsyncRequestBody requestProvider) {
```
Comment (shorea) [inline codegen/src/test/resources/software/amazon/awssdk/codegen/poet/client/test-json-async-client-interface.java]: param name

File: core/src/main/java/software/amazon/awssdk/core/async/AsyncRequestBody.java [hunk 9]
```diff
@@ -41,49 +41,48 @@
...
-        return new SingleByteArrayAsyncRequestProvider(string.getBytes(cs));
+    static AsyncRequestBody fromString(String string, Charset cs) {
+        return new SingleByteArrayAsyncRequestBody(string.getBytes(cs));
     }
 
     /**
-     * Creates an {@link AsyncRequestProvider} that uses a single string as data with UTF_8 encoding.
+     * Creates an {@link AsyncRequestBody} that uses a single string as data with UTF_8 encoding.
      *
      * @param string The string to provider.
```
Comment (shorea) [inline core/src/main/java/software/amazon/awssdk/core/async/AsyncRequestBody.java]: lol this sounds silly

File: core/src/main/java/software/amazon/awssdk/core/client/AsyncClientHandlerImpl.java [hunk 10]
```diff
@@ -126,9 +126,9 @@
         runAfterMarshallingInterceptors(executionContext);
         SdkHttpFullRequest marshalled = runModifyHttpRequestInterceptors(executionContext);
 
-        SdkHttpRequestProvider requestProvider = executionParams.getAsyncRequestProvider() == null
+        SdkHttpRequestProvider requestProvider = executionParams.getAsyncRequestBody() == null
```
Comment (shorea) [inline core/src/main/java/software/amazon/awssdk/core/client/AsyncClientHandlerImpl.java]: param name

File: core/src/test/java/software/amazon/awssdk/core/async/AsyncRequestBodyTest.java [hunk 11]
```diff
@@ -27,11 +27,11 @@
 import org.reactivestreams.Subscriber;
 import software.amazon.awssdk.http.async.SimpleSubscriber;
 
-public class AsyncRequestProviderTest {
+public class AsyncRequestBodyTest {
 
     @Test
     public void canCreateAStringProvider() {
```
Comment (shorea) [inline core/src/test/java/software/amazon/awssdk/core/async/AsyncRequestBodyTest.java]: provider

**Comment by millems** [comment]
Updated to address comments!

**Pull request merged**